### PR TITLE
Adding Client side error handling for Branch View

### PR DESCRIPTION
**Pull request by sojanpr**
Branch view is displayed only if there is  no error while loading the
branch view URL.

Added handling for http error

@aaustin @Sarkar @EvangelosG @pbcorriganBranch

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/ServerRequestActionCompleted.java [hunk 1]
```diff
@@ -68,9 +66,11 @@ public void onRequestSucceeded(ServerResponse resp, Branch branch) {
                     if (post != null && post.has(Defines.Jsonkey.Event.getKey())) {
                         actionName = post.getString(Defines.Jsonkey.Event.getKey());
                     }
-                    Activity currentActivity = Branch.getInstance().currentActivityReference_.get();
-                    JSONObject branchViewJsonObj = resp.getObject().getJSONObject(Defines.Jsonkey.BranchViewData.getKey());
-                    BranchViewHandler.getInstance().showBranchView(branchViewJsonObj, actionName, currentActivity, callback_);
+                    if(Branch.getInstance().currentActivityReference_ != null) {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/ServerRequestActionCompleted.java]: add a space here after `if`

**Comment by aaustin** [comment]
nice 👍  looks good

**Comment by sojanpr** [comment]
Since HTTP errors are not reported by certain devices even with api level 23 , falling back to grabbing html for the url and loading it on the web view in case the view is specified by url. Can we redo a review on this PR?

File: Branch-SDK/src/io/branch/referral/BranchViewHandler.java [hunk 2]
```diff
@@ -138,36 +129,82 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
...
+                branchView.updateUsageCount(currentActivity.getApplicationContext(), branchView.branchViewID_);
+                parentActivityClassName_ = currentActivity.getClass().getName();
+
+                RelativeLayout layout = new RelativeLayout(currentActivity);
+                layout.setVisibility(View.GONE);
+                RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+                layoutParams.setMargins(3, 3, 3, 3);
+
+                layout.addView(webView, layoutParams);
+                layout.setBackgroundColor(Color.parseColor("#11FEFEFE"));
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchViewHandler.java]: Can you paste in a screenshot of the webview? I'm not super comfortable with us hardcoding some of this display stuff. Must we add a margin/background?

Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchViewHandler.java]: (I know this was there before)

File: Branch-SDK/src/io/branch/referral/BranchViewHandler.java [hunk 3]
```diff
@@ -138,36 +129,82 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
...
 
                 @Override
+                public void onPageStarted(WebView view, String url, Bitmap favicon) {
+                    super.onPageStarted(view, url, favicon);
+                }
+
+                @Override
+                public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+                    super.onReceivedError(view, errorCode, description, failingUrl);
+                    webViewLoadError_ = true;
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/BranchViewHandler.java]: You said this happened regularly? Do we need to handle dismissal and callback to the dev when this occurs?

File: Branch-SDK/src/io/branch/referral/BranchViewHandler.java (hunk 2 above)
Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/BranchViewHandler.java]: Removed the margin added to Branch View

File: Branch-SDK/src/io/branch/referral/BranchViewHandler.java (hunk 3 above)
Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/BranchViewHandler.java]: `onReceivedError` is always followed by `onPageFinished` There we already have the callback to dev and Branch view Showing status variable update

**Pull request merged**